components/arm/universalrobots/ur5e_test.go

```diff
@@ -435,3 +437,84 @@ func TestArmReconnection(t *testing.T) {
 	childCancel()
 	_ = ua.Close(ctx)
 }
+
+func TestUpdateAction(t *testing.T) {
+	var remote atomic.Bool
+
+	remote.Store(false)
+
+	statusBlob, err := os.ReadFile("armBlob")
+	test.That(t, err, test.ShouldBeNil)
+
+	logger := golog.NewTestLogger(t)
+	parentCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ctx, childCancel := context.WithCancel(parentCtx)
+
+	closer, _, _, err := setupListeners(ctx, statusBlob, &remote)
+	test.That(t, err, test.ShouldBeNil)
+
+	cfg := config.Component{
+		Name: "testarm",
+		ConvertedAttributes: &AttrConfig{
+			Speed:               0.3,
+			Host:                "localhost",
+			ArmHostedKinematics: false,
+		},
+	}
+
+	shouldNotReconfigureCfg := config.Component{
+		Name: "testarm",
+		ConvertedAttributes: &AttrConfig{
+			Speed:               0.5,
+			Host:                "localhost",
+			ArmHostedKinematics: false,
+		},
+	}
+
+	shouldReconfigureCfg := config.Component{
+		Name: "testarm",
+		ConvertedAttributes: &AttrConfig{
+			Speed:               0.5,
+			Host:                "new",
+			ArmHostedKinematics: false,
+		},
+	}
+
+	injectRobot := &inject.Robot{}
+
+	ur5e, err := URArmConnect(parentCtx, injectRobot, cfg, logger)
```

Comment:
Won't this fail if this test does not pass in the connection timeout period?

I think this will also work if you do:
```
attrs, ok := cfg.ConvertedAttributes.(*AttrConfig)
test.That(ok is true)
&URArm{
	speed:                    attrs.Speed,
	urHostedKinematics:       attrs.ArmHostedKinematics,
	host:                     attrs.Host,
}
```
and this will avoid attempts to make network calls

Reply:
this is a cool solution to get around making network calls